src/co/elastic/NotificationManager.groovy

```diff
@@ -35,57 +35,55 @@ This method generates flakey test data from Jenkins test results
  * @param secret Vault path to secrets which hold authentication information for Elasticsearch
  * @param jobInfo JobInfo data collected from job-info.json
  * @param testsErrors list of test failed, see src/test/resources/tests-errors.json
- * @param flakyReportIdx, what's the id.
- * @param flakyThreshold, to tweak the score.
  * @param testsSummary object with the test results summary, see src/test/resources/tests-summary.json
  * @param querySize The maximum value of results to be reported. Default 500
  * @param queryTimeout Specifies the period of time to wait for a response. Default 20s
  * @param disableGHComment whether to disable the GH comment notification.
  * @param disableGHIssueCreation whether to disable the GH create issue if any flaky matches.
+ * @param jobName
 */ 
 def analyzeFlakey(Map args = [:]) {
     def es = args.containsKey('es') ? args.es : error('analyzeFlakey: es parameter is required')
+    def jobName = args.containsKey('jobName') ? args.jobName : error('analyzeFlakey: jobName parameter is required')
     def secret = args.containsKey('es_secret') ? args.es_secret : null
-    def flakyReportIdx = args.containsKey('flakyReportIdx') ? args.flakyReportIdx : error('analyzeFlakey: flakyReportIdx parameter is required')
     def testsErrors = args.containsKey('testsErrors') ? args.testsErrors : []
-    def flakyThreshold = args.containsKey('flakyThreshold') ? args.flakyThreshold : 0.0
     def testsSummary = args.containsKey('testsSummary') ? args.testsSummary : null
     def querySize = args.get('querySize', 500)
-    def queryTimeout = args.get('queryTimeout', '20s')
     def disableGHComment = args.get('disableGHComment', false)
     def disableGHIssueCreation = args.get('disableGHIssueCreation', false)
-
     def labels = 'flaky-test,ci-reported'
     def boURL = getBlueoceanDisplayURL()
     def flakyTestsWithIssues = [:]
     def genuineTestFailures = []
 
-    if (!flakyReportIdx?.trim()) {
-      error 'analyzeFlakey: did not receive flakyReportIdx data'
-    }
-
-    // Only if there are test failures to analyse
-    if(testsErrors.size() > 0) {
+    // 1. Only if there are test failures to analyse
+    // 2. Only continue if we have a jobName passed in. This does not raise an error to preserve
+    // backward compatability.
+    if(testsErrors.size() > 0 && jobName != null) {
```

Comment:
L47 will throw an error, so I guess the jobName will never be null but maybe empty (https://github.com/elastic/apm-pipeline-library/pull/1016/files#diff-e79413d8e910494d1fa40002bd01bb66c3830ab0646a68e93e15c4d43a7a22e7R69)?

```suggestion
    if(testsErrors.size() > 0 && jobName?.trim()) {
```